frontend/src/js/components/modals/TemplateEditor.jsx

```diff
@@ -1,10 +1,15 @@
 import React from 'react'
 import TemplatePreview from '../TemplatePreview.jsx'
+import TemplateLabelsEditor from '../editors/template/TemplateLabelsEditor.jsx'
 
 
 var TemplateEditor = React.createClass({
     propTypes: {
-        template: React.PropTypes.object
+        template: React.PropTypes.object,
+        updateTemplate: React.PropTypes.func,
+    },
+    onUpdate(attribute, key, value) {
+        this.props.updateTemplate(this.props.template.name, attribute, key, value);
```

Comment:
Note that you only need to do that in this file, and everything that happens internally to the subcomponents should be able to stay the same.